flowmap/formats/ugrid.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import pathlib
+
+# TODO, switch to pyugrid after next release
+import netCDF4
+import numpy as np
+
+from tvtk.api import tvtk
+
+from .formats import transform
+from .netcdf import NetCDF
+from .. import particles
+
+logger = logging.getLogger(__name__)
+
+
+class UGrid(NetCDF):
+    """UGrid NetCDF format"""
+
+    def validate(self):
+        """validate a file"""
+        valid = True
+
+        with netCDF4.Dataset(self.path) as ds:
+            variables = ds.variables.keys()
+        for var in ("mesh2d_ucx", "mesh2d_ucy", "mesh2d_face_nodes", "mesh2d_node_x", "mesh2d_node_x"):
+            if var not in variables:
+                logger.warn(
+                    "%s not found in variables of file %s",
+                    var,
+                    self.path
+                )
+                valid = False
+                return valid
+        return valid
+
+    @property
+    def grid(self):
+        with netCDF4.Dataset(self.path) as ds:
+            x = ds.variables['mesh2d_node_x'][:]
+            y = ds.variables['mesh2d_node_y'][:]
+            faces = ds.variables['mesh2d_face_nodes'][:]
+        z = np.zeros_like(x)
+        points = np.c_[x, y, z]
+        grid = dict(
+            x=x,
+            y=y,
+            z=z,
+            points=points,
+            faces=faces
+        )
+        return grid
+
+    def to_polydata(self):
+        grid = self.grid
+
+        faces = grid['faces']
+        points = grid['points']
+
+        n_cells = faces.shape[0]
+        points.shape, n_cells
+
+        cell_array = tvtk.CellArray()
+
+        assert not hasattr(faces, 'mask'), 'should not be a masked array'
+        assert faces.shape[1] == 4, 'we expect quads only'
```

Comment:
Couldn't this be directly solved by:
n_edges = faces.shape[1] 
counts = np.ones((n_cells), dtype=faces.dtype) * n_edges

Or will this go wrong later on in the script?

Reply:
I hardcoded this for now, documented.